operator/controllers/istio_controller.go

```diff
@@ -41,11 +45,34 @@ func TemplateRateLimiter(failureBaseDelay time.Duration, failureMaxDelay time.Du
 		&workqueue.BucketRateLimiter{Limiter: rate.NewLimiter(rate.Limit(frequency), burst)})
 }
 
-//+kubebuilder:rbac:groups=operator.kyma-project.io,resources=istios,verbs=get;list;watch;create;update;patch;delete
-//+kubebuilder:rbac:groups=operator.kyma-project.io,resources=istios/status,verbs=get;update;patch
-//+kubebuilder:rbac:groups=operator.kyma-project.io,resources=istios/finalizers,verbs=update
-//+kubebuilder:rbac:groups="",resources=namespaces,verbs=get;create;update;patch
+func NewReconciler(mgr manager.Manager) *IstioReconciler {
+	return &IstioReconciler{
+		Client: mgr.GetClient(),
+		Scheme: mgr.GetScheme(),
+	}
+}
+
+func (r *IstioReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
+
+	logger.Info("Was called to reconcile Kyma Istio Service Mesh")
+
+	istioCR := operatorv1alpha1.Istio{}
+	if err := r.Client.Get(ctx, req.NamespacedName, &istioCR); err != nil {
+		logger.Error(err, "Error during fetching Istio CR")
+	}
+
+	if err := install.ReconcileIstio(&istioCR); err != nil {
+		logger.Error(err, "Error occurred during reconciliation of Istio Operator")
+	}
+
+	return ctrl.Result{}, nil
```

Comment:
https://github.tools.sap/xf-goat/adr/blob/main/doc/adr/0003-istio-operator-design.md#reconciliation-interval
```suggestion
	return ctrl.Result{RequeueAfter: time.Minute*5}, nil
```

Reply:
Good point!